ui/app/components/topo-viz.js

```diff
@@ -144,6 +153,19 @@ export default class TopoViz extends Component {
     if (this.args.onNodeSelect) this.args.onNodeSelect(this.activeNode);
   }
 
+  @action showTooltip(allocation, element) {
+    const bbox = element.getBoundingClientRect();
+    this.highlightAllocation = allocation;
+    this.tooltipProps = {
+      left: window.visualViewport.pageLeft + bbox.left + bbox.width / 2,
+      top: window.visualViewport.pageTop + bbox.top,
```

Comment:
I hadn’t heard of `visualViewport`! Sadly at the moment it’s feature flagged in Firefox, so it didn’t work for me until I tried it in Safari and Chrome 😞

<img width="1132" alt="image" src="https://user-images.githubusercontent.com/43280/97881438-05c94900-1ce8-11eb-92f4-66c01d7880b6.png">

Reply:
Curse the internet's betrayal of my trust. I'll swap this out with an older approach.